### PR TITLE
add on_click for music block + add `id` field for rotating text

### PR DESCRIPTION
**Pull request by Dieterbe**
couldn't quite figure out the best way to pass the id into the rotatingtext widget. but this does the job. i'm new to rust :)

### CODE REVIEW

File: blocks.md [hunk 1]
```diff
@@ -701,6 +701,7 @@ Key | Values | Required | Default
 `separator` | String to insert between artist and title | No | `" - "`
 `buttons` | Array of control buttons to be displayed. Options are prev (previous title), play (play/pause) and next (next title) | No | `[]`
 `on_collapsed_click` | Shell command to run when the music block is clicked while collapsed. | No | None
+`on_click` | Command to execute when the button is clicked while not collapsed. | No | None
```
[the remark this answers is not in the document]
Comment (Dieterbe) [reply blocks.md:704]: Not sure. I see that block and button are used interchangeably in this file. e.g. for the cpu usage block on_click also refers to a button.
should it be block everywhere consistently? I could do that in a followup PR.

[the remark this answers is not in the document]
Comment (ammgws) [reply blocks.md:704]: The music block already has clickable widgets (play-pause/prev/next), so the wording is a bit confusing.

File: src/blocks/music.rs [hunk 2]
```diff
@@ -407,6 +418,13 @@ impl Block for Music {
                     let command = self.on_collapsed_click.as_ref().unwrap();
                     spawn_child_async("sh", &["-c", command])
                         .block_error("music", "could not spawn child")?;
+                } else {
+                    if event.matches_name(self.id()) {
+                        if let Some(ref cmd) = self.on_click {
+                            spawn_child_async("sh", &["-c", cmd])
+                                .block_error("cpu", "could not spawn child")?;
```
Comment (ammgws) [inline src/blocks/music.rs:425]: copy paste error? "cpu"

File: blocks.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dieterbe) [reply blocks.md:704]: ok, will change

**Comment by Dieterbe** [comment]
hi @ammgws i have addressed the feedback.
thanks!

**Comment by ammgws** [comment]
Thanks!

**Pull request merged**